src/containerapp/azext_containerapp/custom.py

```diff
@@ -1257,9 +1268,9 @@ def containerapp_up(cmd,
         target_port = 80 if not target_port else target_port
 
     if image:
-        if ingress and not target_port:
-            target_port = 80
-            logger.warning("No ingress provided, defaulting to port 80. Try `az containerapp up --ingress %s --target-port <port>` to set a custom port.", ingress)
+        if ingress and not target_port and target_port != 0:
+            target_port = 0
+            logger.warning("No target-port provided, defaulting to auto-detect. Try `az containerapp up --ingress %s --target-port <port>` to set a custom port.", ingress)
```

Comment:
Will this change lead to a breaking change?

Reply:
No, the auto-detection will still cover the old scenario where the app was listening on port 80.

Reply:
Sounds good